### PR TITLE
Fhammerl/container hooks run script step

### PR DESCRIPTION
**Pull request by fhammerl**

### CODE REVIEW

File: src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public interface IContainerHookManager : IRunnerService
         Task PrepareJobAsync(IExecutionContext context, List<ContainerInfo> containers);
         Task CleanupJobAsync(IExecutionContext context, List<ContainerInfo> containers);
         Task ContainerStepAsync(IExecutionContext context);
-        Task RunScriptStepAsync(IExecutionContext context);
+        Task RunScriptStepOnJobContainerAsync(IExecutionContext context, ContainerInfo container, string arguments, string fileName, IDictionary<string, string> environment, string prependPath, string workingDirectory);
```
Comment (fhammerl) [inline src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs:22]: Rename method `ScriptStepAsync`

File: src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs [hunk 2]
```diff
@@ -108,11 +108,33 @@ public async Task ContainerStepAsync(IExecutionContext context)
...
+                Args = new HookStepArgs
+                {
+                    Container = container.GetHookContainer(),
+                    EntryPointArgs = entryPointArgs.Split(' ').Select(arg => arg.Trim()),
+                    EntryPoint = entryPoint,
+                    EnvironmentVariables = environmentVariables,
+                    PrependPath = prependPath,
+                    WorkingDirectory = workingDirectory,
+                },
+                State = JsonUtility.FromString<dynamic>(hookState.ToString())
```
Comment (fhammerl) [inline src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs:130]: Also nullcheck (maybe a method `GetHookState`?)

File: src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs [hunk 3]
```diff
@@ -108,11 +108,33 @@ public async Task ContainerStepAsync(IExecutionContext context)
...
+            };
+
+            var response = await ExecuteHookScript(context, input);
+            if (response != null)
```
Comment (fhammerl) [inline src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs:134]: Can `response.State` be null?

File: src/Runner.Worker/Handlers/StepHost.cs [hunk 4]
```diff
@@ -230,10 +219,6 @@ public async Task<string> DetermineNodeRuntimeVersion(IExecutionContext executio
                                             bool inheritConsoleHandler,
                                             CancellationToken cancellationToken)
         {
-            // make sure container exist.
-            ArgUtil.NotNull(Container, nameof(Container));
```
Comment (fhammerl) [inline src/Runner.Worker/Handlers/StepHost.cs:234]: Leave arg checks if runner is NOT using `containerHooks`?

File: src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs (hunk 2 above)
Comment (ruvceskistefan) [reply src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs:130]: I'll make a method `GetHookState` and just call it.

File: src/Runner.Worker/Handlers/StepHost.cs (hunk 4 above)
Comment (ruvceskistefan) [reply src/Runner.Worker/Handlers/StepHost.cs:234]: Yes, I think we should leave arg cheks in the case when the runner isn't using container hooks.

**Pull request merged**